demonstrations/tutorial_learning_dynamics_incoherently.py

```diff
@@ -0,0 +1,437 @@
+# :doc:`variational classifiers demo </demos/tutorial_variational_classifier>`.
+#
+# .. note ::
+#
+#    We use *local* measurements to keep the computational complexity low and
+#    because classical shadows are well-suited to estimating local observables [#Jerbi]_.
+#    For this reason, the following circuit returns local density matrices for each qubit. In
+#    hardware, the density matrix is obtained via state tomography using Pauli measurements or classical shadows.
+
+
+@qml.qnode(dev)
+def model_circuit(params, random_state):
+    qml.StatePrep(random_state, wires=range(n_qubits))
+    # Parameterized quantum circuit with the same gate structure as the target
+    for i in range(n_qubits):
+        qml.RX(params[i], wires=i)
+
+    for i in reversed(range(n_qubits - 1)):
+        qml.IsingZZ(params[n_qubits + i], wires=[i, i + 1])
+    return [qml.density_matrix(i) for i in range(n_qubits)]
+
+
+initial_params = pnp.random.random(size=n_qubits*2-1, requires_grad=True)
+
+qml.draw_mpl(model_circuit)(initial_params, random_states[0])
+plt.show()
+
+######################################################################
+# 5. Training a model circuit using classical shadows in a cost function
+# ------------------------------------------------------------------------
+#
+# We now have to find the optimal parameters for ``model_circuit`` to mirror the ``target_circuit``.
+# We can estimate the similarity between the circuits according to this cost function (see
+# Appendix B of [#Jerbi]_):
+#
+# .. math:: C^l_N(\theta) = 1 - \frac{1}{nN}\sum^N_{j=1}\sum^n_{i=1}Tr[U|\psi^{(j)}\rangle\langle\psi^{(j)}|U^\dagger O^{(j)}_i(\theta)],
+#
+# where :math:`n` is the number of qubits, :math:`N` is the number of initial states, :math:`\psi^{(j)}`
+# are random states, :math:`U` is our target unitary operation, and :math:`O_i` is the local density
+# matrix for qubit :math:`i` after applying the ``model_circuit``. That is, the local states
+# :math:`\rho_{i}^{(j)}` are used as the observables:
+#
+# .. math:: O_{i}^{(j)}(\theta) := \rho_{i}^{(j)}.
+#
+# We can calculate this cost for our system by using the
+# `shadow measurements <#time-evolution-and-classical-shadow-measurements>`_ to estimate
+# the expectation value of :math:`O_i`. Roughly, this cost function measures the fidelity between
+# the model circuit and the target circuit, by proxy of the single-qubit reduced states
+# :math:`\rho_{i}^{(j)}` of the model over a variety of input-output pairs.
+
+
+def cost(params):
+    cost = 0.0
+    for idx, random_state in enumerate(random_states):
+        # Obtain the density matrices for each qubit
+        observable_mats = model_circuit(params, random_state)
+        # Convert to a PauliSentence
+        observable_pauli = [
+            qml.pauli_decompose(observable_mat, wire_order=[qubit])
+            for qubit, observable_mat in enumerate(observable_mats)
+        ]
+        # Estimate the overlap for each qubit
```

Comment:
```suggestion
        # estimate the overlap for each qubit
```
For consistency. (Note that you can either change these or capitalize other comments in previous codeblocks.)